src/tests/UserInputBlock.test.tsx

```diff
@@ -0,0 +1,144 @@
+
+test('UserInputBlock renders with content, prompt, and form as textarea', async () => {
+  render(
+    <UserInputBlock content="<p>Content text</p>" prompt="<p>Prompt text</p>" ack="<p>Ack text</p>/>"/>
+  )
+
+  screen.getByText('Content text')
+  screen.getByText('Prompt text')
+  screen.getByRole('textbox')
+  expect(document.querySelector('textarea')).not.toBeNull()
+  expect(screen.getByRole('button').textContent).toBe('Submit')
+})
+
+test('UserInputBlock allows setting custom button text', async () => {
+  render(
+    <UserInputBlock content="<p>Content text</p>" prompt="<p>Prompt text</p>" ack="<p>Ack text</p>/>" buttonText="CustomButtonText"/>
+  )
+  expect(screen.getByRole('button').textContent).toBe('CustomButtonText')
+})
+
+test('UserInputBlock does not render if waitForEvent does not fire', async () => {
+  render(
+    <UserInputBlock content="<p>Content text</p>" prompt="<p>Prompt text</p>" ack="<p>Ack text</p>/>" waitForEvent='someEvent'/>
+  )
+
+  expect(screen.queryByText('Content text')).toBeNull()
+  expect(screen.queryByText('Prompt text')).toBeNull()
+  expect(screen.queryByRole('textbox')).toBeNull()
+  expect(screen.queryByRole('button')).toBeNull()
+})
+
+test('UserInputBlock does render if waitForEvent is fired', async () => {
+  render(
+    <UserInputBlock content="<p>Content text</p>" prompt="<p>Prompt text</p>" ack="<p>Ack text</p>/>" waitForEvent='someEvent'/>
+  )
+
+  fireEvent(document, new CustomEvent('someEvent'))
+
+  screen.getByText('Content text')
+  screen.getByText('Prompt text')
+  screen.getByRole('textbox')
+  screen.getByRole('button')
+})
+
+test('UserInputBlock removes and adds expected content on valid submission', async () => {
+  render(
+    <UserInputBlock content="<p>Content text</p>" prompt="<p>Prompt text</p>" ack="<p>Ack text</p>/>"/>
+  )
+
+  fireEvent.change(screen.getByRole('textbox'), { target: { value: 'Input text' } })
+  await act(async () => {
+    screen.getByRole('button').click()
+  })
+
+  screen.getByText('Ack text')
+  expect(screen.queryByText('Prompt text')).toBeNull()
+  expect(screen.queryByRole('textbox')).toBeNull()
+  expect(screen.queryByRole('button')).toBeNull()
+})
+
+test('UserInputBlock requires non-empty input', async () => {
+  render(
+    <UserInputBlock content="<p>Content text</p>" prompt="<p>Prompt text</p>" ack="<p>Ack text</p>/>"/>
+  )
+
+  await act(async () => {
+    screen.getByRole('button').click()
+  })
+
+  screen.getByText('Content text')
+  screen.getByText('Prompt text')
+  screen.getByRole('textbox')
+  screen.getByRole('button')
+  screen.getByText('Please provide valid input')
+
+  await act(async () => {
+    fireEvent.change(screen.getByRole('textbox'), { target: { value: '   ' } })
+  })
+
+  screen.getByText('Please provide valid input')
+
+  await act(async () => {
+    fireEvent.change(screen.getByRole('textbox'), { target: { value: 'S' } })
+  })
+  expect(screen.queryByText('Please provide valid input')).toBeNull()
+})
```

Comment:
Why do you use this async function to fire the click and change events?

Reply:
Ah -- you can try this locally, but if you remove that wrapper you'll get `Warning: An update to Formik inside a test was not wrapped in act(...).`. See related upstream issue [here](https://github.com/jaredpalmer/formik/issues/1543). This was my fix for addressing that in the corresponding test cases.